src/vm/functions/assets.rs

```diff
@@ -0,0 +1,211 @@
+use std::convert::TryFrom;
+
+use vm::functions::tuples;
+use vm::functions::tuples::TupleDefinitionType::{Implicit, Explicit};
+
+use vm::types::{Value, OptionalData, BuffData, PrincipalData, BlockInfoProperty, AtomTypeIdentifier};
+use vm::representations::{SymbolicExpression};
+use vm::errors::{Error, UncheckedError, InterpreterError, RuntimeErrorType, InterpreterResult as Result, check_argument_count};
+use vm::{eval, LocalContext, Environment};
+
+pub fn special_mint_token(args: &[SymbolicExpression],
+                          env: &mut Environment,
+                          context: &LocalContext) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let token_name = args[0].match_atom()
+        .ok_or(UncheckedError::InvalidArgumentExpectedName)?;
+
+    let amount = eval(&args[1], env, context)?;
+    let to =     eval(&args[2], env, context)?;
+
+    if let (Value::Int(amount),
+            Value::Principal(ref to_principal)) = (amount, to) {
+        if amount < 0 {
```

Comment:
Should this be `amount <= 0`?  Do we support transferring 0 tokens?

Reply:
I don't think we should support transferring 0 tokens

Reply:
We can add an error code response for this though rather than aborting the entire transaction, though.

Reply:
I think first class assert support highlights the need for uint support, tagging https://github.com/blockstack/blockstack-core/issues/974